src/blocking_retry.rs

```diff
@@ -105,8 +105,8 @@ where
     ///     Ok(())
     /// }
     /// ```
-    pub fn when(mut self, retryable: fn(&E) -> bool) -> Self {
-        self.retryable = retryable;
+    pub fn when(mut self, retryable: impl FnMut(&E) -> bool + 'static) -> Self {
```

Comment:
This change will add two new box in retry which seems not a good idea to me. Can we add some context in retry instead? For example:

```rust
do()
  .retry()
  .context(id) // Set context for retry
  .when_with(|err, ctx| true) // Set when with context
  .notify_with(|err, dur, ctx| {}) // Set notify with context
  .await
```

Reply:
That's interesting, I'll try to experiment the idea, but that would be a new PR since it's a completely different approach.